archeryutils/handicaps/tests/test_handicaps.py

```diff
@@ -319,6 +319,271 @@ def test_array(self) -> None:
 
         np.testing.assert_allclose(sigma_r_array, sigma_r_expected_array)
 
+class TestSBar:
+    """
+    Class to test generic implemtation of s_bar (mean arrow score)
+
+    Comparision tests calculate average arrow score using the existing hardcoded
+    implementation for supported target faces, and checks that the generic implementation
+    from a raw dictionary of target parameters gives the same result (to within
+    floating point accuracy).
+
+
+    Methods
+    -------
+    test_compare_vs_40cm_compound()
+        Checks a 5 ring 10 zone face scoring x for 10
+    test_compare_vs_worcester()
+        Checks a 16 inch 5 zone worcster face
+    test_compare_vs_imperial_122()
+        Checks a 5 zone scoring 122 face
+    test_compare_vs_hit_miss()
+        Checks a 60mm hit miss disc
+    test_compare_vs_80cm_small()
+        Checks a 6 ring 80cm face
+    """
+    def test_empty_spec(self):
+        """
+        Check expected score is zero when no target rings are defined.
+        """
+        s_bar = hc_eq.s_bar({}, 0.01, 0.1)
+        assert s_bar == 0
+
+    def test_unsorted_spec(self):
+        """
+        Check expected score is insensitive to order of input spec.
+        """
+        s_bar_reversed_spec = hc_eq.s_bar({0.2: 2, 0.1: 1}, 0.01, 0.1)
+        s_bar = hc_eq.s_bar({0.1: 1, 0.2: 2}, 0.01, 0.1)
+
+        assert s_bar_reversed_spec == s_bar
```

Comment:
Nice!

Reply:
Needed to beef this test up as it actually happened to work for reversed keys but not for a randomly ordered spec, sorted now though.